File: colorize_powershell_like.py
from colorama import init, Fore, Style
from sys import argv
from re import sub
init(autoreset=True)


def colorize(string):
    string = sub(
        r'<BOLD:(.+?)>(.+?)($|<)',
        Style.BRIGHT + r'<\1>\2' + Style.RESET_ALL + r'\3',
        string
    )
    string = sub(r'<PURPLE>', Fore.CYAN, string)
    string = sub(r'<GREEN>', Fore.GREEN, string)
    string = sub(r'<WHITE>', Fore.WHITE, string)
    string = sub(r'<CYAN>', Fore.CYAN, string)
    string = sub(r'<RED>', Fore.RED, string)
    string = sub(r'<YELLOW>', Fore.YELLOW, string)
    string = sub(r'<BLUE>', Fore.BLUE, string)
    string = sub(r'<MAGNENTA>', Fore.MAGENTA, string)

    # Symbol substitutions (will substitute in shell land)
    # string = sub(r'<SYM:UP>', '<U>', string)
    # string = sub(r'<SYM:DOWN>', '<D>', string)
    # string = sub(r'<SYM:ADD>', '<A>', string)
    # string = sub(r'<SYM:CHG>', '<C>', string)
    # string = sub(r'<SYM:UNTRACKED>', '<UT>', string)

    return string


if __name__ == '__main__':
    print(
        colorize(' '.join(argv[1:])).replace('\x1b', '\\x1b')
    )

File: colorize_powershell_like.go
package main

import (
	"fmt"
	"os"
	"regexp"
	"strings"
)

var IRColorSymMapPowershell = map[string]string{
	"<*BOLD>":   "\u001b[1m",
	"<PURPLE>":  "\u001b[36m",
	"<GREEN>":   "\u001b[32m",
	"<WHITE>":   "\u001b[37m",
	"<CYAN>":    "\u001b[36m",
	"<RED>":     "\u001b[31m",
	"<YELLOW>":  "\u001b[33m",
	"<BLUE>":    "\u001b[36m", // "\u001b[34m", (too dark!)
	"<MAGENTA>": "\u001b[35m",
	"<RESET>":   "\u001b[0m",
	/* Symbol substitutions (will substitute in shell land)
	"<SYM:UP>": "↑",
	"<SYM:DOWN>": "↓",
	"<SYM:ADD>": "+",
	"<SYM:CHG>": "Δ",
	"<SYM:UNTRACKED>": "…",
	*/
}

func colorizePowershell(rawString string) string {
	// Convert style types to their ascii prefixes
	boldRegex := regexp.MustCompile(`<BOLD:(.+?)>(.+?)($|<)`)
	rawString = boldRegex.ReplaceAllString(rawString, `<*BOLD><$1>$2<RESET>$3`)

	for ir, asciiCode := range IRColorSymMapPowershell {
		rawString = strings.Replace(rawString, ir, asciiCode, -1)
	}

	// Auto revert back to normal colors
	return IRColorSymMapPowershell["<RESET>"] + rawString + IRColorSymMapPowershell["<RESET>"]
}

func main() {
	fmt.Println(
		strings.Replace(
			colorizePowershell(strings.Join(os.Args[1:], " ")),
			"\x1b",
			"\\x1b",
			-1,
		),
	)
}
